docs/requestAwareMiddleware.md

```diff
@@ -0,0 +1,45 @@
+# The "Request Aware" middleware.
+
+Some middleware (such as the "static" middleware included in this package) do not keep any kind of state variables that
```

Comment:
Actually, that's not the case.  The `counter` middleware used in the tests is an example of middleware that needs to keep state variables between requests.  You might also have middleware that is coordinating state between users, such as in a multiplayer game or shared editing environment.